src/lib/Components/Bidding/Components/Timer.tsx

```diff
@@ -93,25 +99,28 @@ function currentTimerState({ isPreview, isClosed, liveStartsAt }: Props) {
   }
 }
 
-interface CountdownProps {
+export interface CountdownProps {
   duration: moment.Duration
-  label: string
+  label?: string
+  [propName: string]: {}
```

Comment:
This is an attempt to also accept arbitrary props that need to be passed from the wrapper component.

Reply:
Why is there a need for arbitrary props?

Reply:
In this case, the component that's being wrapped needs a relay prop `CommercialInformation_artwork` passed to it  (but it could be anything).

Reply:
[Here’s a solution](https://www.typescriptlang.org/play/index.html?jsx=2#code/PQKhCgAIUgVALAppA5gewIYBtIEsDOkALmpBgMbmIAORZkASohXeWgLbVoB2i3dRAJ7VkGQhkjUATmmqQAFAGEArvhLtFaZfwAmaAO7dNnHnyIBKYvAxEoMfPC1YdZSjTrTZhfCPK4AZriILgBGgmTcePyIUv4UyEpaugZGHFy8-AAKMtT4ltRi+MGQYVaIdpIYUmYAdJAAgjo6uES4PNhY4URIldX8kGwmGXQOTi4F+PgAtN0yyijwEeH+ylLdMZI5hN02FeuQKmocmtpEeobG6WauVLT4NXbA4OC4JmuMzOR0-jLskABE1RY-2efGUf3qyi+bW4sFeMQAykQbMgAN5QSCZBgAUQAagBJbEAdUgAF4AVi8YSif8ADQYgAy+Nx2IA+viAHKwbEAcQY9Vg+IA8hzWQBVTKKIUAWU5PLJAKZLPZXN5-MFIvFkplcrpjOZbM53L5AuFopFPKFcoV-yVhtVJo15o5lt19MgBwZQoR1vJ-0UXp9Lr1HoD3uxABEbWGEZG9QBfZ64aKxeIHJJnFKXUxZLaQdEenSrGwwgBckG44JCMQxWAw1awAH5y2opMmUBjWuxEciiIhy5DoTw4d2pEiUeBE+AhCJIOO+9KMNwMCgYtkvAAeDGgCAe6CQAASMWQ+mQOkQ5Dr1SsNjKqD4MVw5EgAANDuoTskLmkc0R17kX02LxIFGZRnAxfdk0vZRzzvNQUUgdglxXGIAHJCDfDNzlSIYzH-fAXwePcQCeUNVA-LCsx-YZ8MgRAAA8+24HRCE-TNv1w3MvHAAA+BUC0gHcIJgCNEECXg7zfcjjkojirn4QCSBKUQPhYAZqOufQWngYTX3fGTTmw7MaK2QjdN0oV-ArNA+1pSBtCwRBJj06SNFknD5L-Uy8EIBjqCwJ8Wk6EDfACIIdDs7oCB8spdP8tN9CQa9YGERAEXINtaEgLSsBwZN-A2RLb32GdkCwNA0AAa0IALKtEZiyEmXAUG4QgWmIUh1l0qSjjcwyqM4ryvEU1Legwbs+ykIiPRIjF9L6r8PN-csmBYGpjLMFKRA3ea2KMjSuNyHjJ0gAAybcwF0+osHwUgKFuOgl3CTxchvR7r2TFzer2gbPPwwCGIeyB-DQKQrDQIpup+uTf3+uzEqfRYqmQfYXu2aw6BynBq0qSZghmeA5gWaboFIyAhXYFodtc6GlpMrw7LqwQ0Cs2mNsO-BjvAS9CjnXtEEXZdVykLcyO+9z2aG16GKYlj036mH6dyXi6MYvg5dWr51oOohRb5lFBZQqR8Op8WFbpvCtmOviBL6c8pHkSwBI9IS9xmuBUvSzLWCXQBMAjoapbqwAA3ZByFcoDXqS5BPqiwhV14Ntn3iqhIEERAiDskJlEe9GCAg93ovwV5-PCCQJmmWYtAWbLRBu0g9G4f3ICBXLwnIZGyBCLQBB6ZNmmD3Ai2wSP7kgIlyjd-ceGCjvr3rHvwaKbLo9fecBeQ4XAOi6t23soodBJmayY9NhWroVEDhpiWdbsmp77RyB4zJAuPTjmpH7EezuEq7gUka-WC5N5ri2DtG+g18JczdmfNQkA6wNhtLdbscD6yICwCCaBPBYFFikCWHgCoAAsAAmAuMCRj8wVIOVow54Rjn5utQMcpnhu2qEQVYkR5Cv0gKbCi5tJZcI9PAtBpJURCKwImKee4cF4O4CI6R1DuASMkcQWh68RHwT7EoyRqJ74fy2FovcwBjpu3MBiRMU5SrpnYKOPw2B8TcBBlIJCCjWRVCIPoUGlVowcBsbgOxDjQbOJhK4tYHipCVRBNzLBdAEQcEQLtcBnkVqfCIDUBEAAxRQes2Y61oqdfMZBQmePLMYXx-jHFBJ4CE9xnin4q3JI-UkNsMSsPYQoAuG4B7GKnqiNGNQxHPymPmPp8iYSDOGVsGoXYewogMRuYAXSMSmKnBudehthZzWvnwnWIjYndgSdswaSi3FhMqiI-0PiYi2KwPYipMjqmnP+ImIx4AgA) that makes `StateManager` generic in a way that it verifies the given countdown component takes the expected props _as well_ as verifying the additional props that the custom countdown component expects is passed to the `StateManager`. Let me know if it needs further explanation and/or if you want to pair!